Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -23,20 +26,45 @@ public SubmissionsDataService(DbContext db)
             .MapCollection<TServiceModel>()
             .FirstOrDefault();
 
-    public Task<IEnumerable<TServiceModel>> GetLatestSubmissions<TServiceModel>(int count)
-        => this.GetQuery(
+    public async Task<IEnumerable<TServiceModel>> GetLatestSubmissions<TServiceModel>(int submissionsPerPage)
+        => await this.GetQuery(
                 orderBy: s => s.Id,
                 descending: true,
-                take: count)
+                take: submissionsPerPage)
             .MapCollection<TServiceModel>()
             .ToEnumerableAsync();
 
-    public async Task<int> GetTotalSubmissionsCount()
-        => await this.GetQuery(
+    public async Task<PagedResult<TServiceModel>> GetLatestSubmissions<TServiceModel>(int submissionsPerPage, int pageNumber)
+            => await this.GetQuery(
+                    filter: s => !s.IsDeleted,
+                    orderBy: s => s.Id,
+                    descending: true)
+                .MapCollection<TServiceModel>()
+                .ToPagedResultAsync(submissionsPerPage, pageNumber);
+
+    public async Task<PagedResult<TServiceModel>> GetAllUnprocessedByUser<TServiceModel>(
+        UserInfoModel user,
+        int submissionsPerPage,
+        int pageNumber)
+    {
+        var submissionsQuery = this.GetQuery(
+                filter: s => !s.Processed && !s.IsDeleted,
                 orderBy: s => s.Id,
-                descending: true)
-            .Select(s => s.Id)
-            .FirstOrDefaultAsync();
+                descending: true);
+
+        if (user.IsLecturer)
+        {
+            submissionsQuery = submissionsQuery.Where(s => s.Participant!.UserId == user.Id);
+        }
```

Comment:
Why filter submissions of the lecturer only?

Reply:
Ok, reply again when it's done

Reply:
Done